streamer/src/nonblocking/quic.rs

```diff
@@ -1152,7 +1152,13 @@ pub mod test {
         for _ in 0..num_bytes {
             s1.write_all(&[0u8]).await.unwrap();
         }
-        s1.finish().await.unwrap();
+        let result = s1.finish().await;
+        match result {
+            Err(quinn::WriteError::UnknownStream) => {
```

Comment:
I am new to this codebase, so it might be obvious.
But to me, the change description provides an important reasoning as to why this matching is needed.
As well as why all the other errors are ignored.
I would suggest putting that explanation here, as a comment, rather than in the commit message.